### PR TITLE
[feat] add filter dropdowns

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Modified `src/utils/helpers.ts`:
- Made `timestampStringToDate` function regular export (no longer default, refactored to address this)
- Added `pSBC` function to allow for shading, blending, and converting colors

Added new component: `DropdownFilter` and its dependencies
- `src/components/FilterDropdown/FilterDropdownMenu.tsx`
- `src/components/FilterDropdown/index.tsx`
- `src/components/FilterDropdown/styles.tsx`

Modified `src/app/cases/page.tsx`:
- Added filters
- Used said filters to filter case data

Deleted `src/app/input-text/page.tsx` to clean up testing

Minor edits:
- `src/styles/colors.ts`
    - Added `abaBlue` for brand color
- `src/components/ListingCard/styles.tsx`
    - Adjusted tag padding for better look
- `src/components/ListingCard/index.tsx`
    - Only needs one role tag (interpreter) since attorney is always needed
- Refactored import paths to use `@/`
- Updated `tsconfig.js` target to `ES2015` to allow spread operators

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

image

First filter: number
Second: hover
Third: clicked (menu open)
Fourth: one item changed
Fifth: multiple items changed/selected

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Review order for `FilterDropdown`:
- `src/components/FilterDropdown/styles.tsx`
- `src/components/FilterDropdown/index.tsx`
- `src/components/FilterDropdown/FilterDropdownMenu.tsx`

Then review `src/app/cases/page.tsx`.

Review other files in any order.

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Add checkboxes to multiselect filters

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

I obviously did NOT come up with the helper functions in `utils`, linked below are the sources for code and inspiration:
- [GitHub Wiki](https://github.com/PimpTrizkit/PJs/wiki/12.-Shade,-Blend-and-Convert-a-Web-Color-(pSBC.js)) explaining `pSBC`

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -18,3 +19,178 @@ export default function timestampStringToDate(ts: string) {
...
+  } else {
+    return false;
+  }
+
+  // HSP equation from http://alienryderflex.com/hsp.html
+  const hsp = Math.sqrt(0.299 * (r * r) + 0.587 * (g * g) + 0.114 * (b * b));
+
+  // Using the HSP value, determine whether the color is light or dark
+  return hsp <= threshold;
+}
```
Comment (varortz) [inline src/utils/helpers.ts]: Could be good to remove this helper for now if it is not being used anywhere

**Comment by david-qing** [comment]
Is it possible to make the filters close if you click on the filter after it opens? as in toggling the dropdown to appear and disappear

**Comment by david-qing** [comment]
also, make sure that these include the new checkboxes that i added to the design. Sorry for this taking so long! as well as maybe trying to include some whitespace in between each item of the multiselect

**Comment by pragyakallanagoudar** [comment]
> also, make sure that these include the new checkboxes that i added to the design. Sorry for this taking so long! as well as maybe trying to include some whitespace in between each item of the multiselect

let's do this in another sprint! you can hold off for rn

**Review by pragyakallanagoudar — APPROVED** [review]
Looks amazing, Jinkang!!! Thank you so much for all your work on this, 2 weeks in the making 🤩🤩

Some next steps are:
* Adding checkboxes to multiselect filters (as @davidqing6432 suggested in his comment)
* Possibly changing around some of the values for the roles and remote vs. in-person filters to make these more clearer? We can discuss this in worksession on Thursday.
* Making sure we have the exact set of filters we need (might be some changes on this after MPH usability testing....also a Thursday conversation!)

**Pull request merged**